### PR TITLE
benchmark workflow & storing it in db

### PR DESCRIPTION
**Pull request by PaulRitter**

### CODE REVIEW

File: .github/workflows/benchmarks.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+on:
+  shedule:
+    - cron: '0 0 * * *'
+  push:
+    tags:
+      - 'v*'
+
+permissions:
+  contents: write
+  deployments: write
```
Comment (PJB3005) [inline .github/workflows/benchmarks.yml]: What do you need these permissions for?

File: .github/workflows/benchmarks.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  ROBUST_BENCHMARKS_SQL_USER: ${{ secrets.BENCHMARKS_WRITE_USER }}
+  ROBUST_BENCHMARKS_SQL_PASSWORD: ${{ secrets.BENCHMARKS_WRITE_PASSWORD }}
+  ROBUST_BENCHMARKS_SQL_DATABASE: benchmarks
+
+jobs:
+  benchmark:
+    name: Run Benchmarks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (PJB3005) [inline .github/workflows/benchmarks.yml:21]: You're not cloning submodules here.

File: .github/workflows/benchmarks.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      - uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '6.0.101'
+      - name: Run benchmark
+        run: cd examples/benchmarkdotnet && dotnet run --filter '*'
```
Comment (PJB3005) [inline .github/workflows/benchmarks.yml]: I think that directory doesn't check out.

File: Robust.Benchmarks/Exporters/SQLExporter.cs [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        try
+        {
+            ctx.Database.Migrate();
+        }
+        finally
+        {
+            ctx.DisposeAsync();
+        }
+
+        ctx.BenchmarkRuns.Add(BenchmarkRun.FromSummary(summary, gitHash));
```
Comment (PJB3005) [inline Robust.Benchmarks/Exporters/SQLExporter.cs]: You're not running `SaveChangesAsync()`.

File: Robust.Benchmarks/Exporters/SQLExporter.cs [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+        }.ConnectionString;
+        builder.UseNpgsql(connectionString);
+        using var ctx = new BenchmarkContext(builder.Options);
+        try
+        {
+            ctx.Database.Migrate();
+        }
+        finally
+        {
+            ctx.DisposeAsync();
```
Comment (PJB3005) [inline Robust.Benchmarks/Exporters/SQLExporter.cs]: You're disposing this asynchronously (and not awaiting) before adding the result?

File: Robust.Benchmarks/Exporters/SQLExporter.cs [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+    public BenchmarkContext() { }
+    public BenchmarkContext(DbContextOptions<BenchmarkContext> options) : base(options) { }
+}
+
+public class BenchmarkRun
+{
+    [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+    public ulong Id { get; set; }
+    public string GitHash { get; set; } = string.Empty;
+    [Column(TypeName = "Date")]
```
Comment (PJB3005) [inline Robust.Benchmarks/Exporters/SQLExporter.cs]: This should be a `timestamptz`.

File: Robust.Benchmarks/Exporters/SQLExporter.cs [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+            {
+                Parameters = r.BenchmarkCase.Parameters.Items.Select(p => new BenchmarkRunParameter
+                {
+                    Name = p.Name,
+                    Value = p.Value
+                }).ToArray(),
+                Statistics = r.ResultStatistics
+            }).ToArray(),
+            Name = summary.BenchmarksCases.First().FolderInfo,
+            RunDate = DateTime.Now,
```
Comment (PJB3005) [inline Robust.Benchmarks/Exporters/SQLExporter.cs]: Store `UtcNow` instead.

File: Robust.Benchmarks/Exporters/SQLExporter.cs [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+
+    private bool TryGetEnvironmentVariable(string name, ILogger logger, [NotNullWhen(true)] out string? value)
+    {
+        value = Environment.GetEnvironmentVariable("name");
+        if (value == null)
+            logger.WriteError($"ROBUST_BENCHMARKS_ENABLE_SQL is set, but {name} is missing.");
+        return value != null;
+    }
+
+    private void Export(Summary summary, ILogger logger)
```
Comment (PJB3005) [inline Robust.Benchmarks/Exporters/SQLExporter.cs:42]: How many times does this method get called for a complete benchmark run? There should only be one DB transaction ideally.

Comment (PaulRitter) [reply Robust.Benchmarks/Exporters/SQLExporter.cs:42]: once

Comment (PaulRitter) [reply Robust.Benchmarks/Exporters/SQLExporter.cs:42]: runs once per benchmark, but out of scope for now

**Pull request merged**